dashboard/app/controllers/api/v1/pd/workshop_enrollments_controller.rb

```diff
@@ -41,7 +41,7 @@ def create
     end
 
     enrollment_email = params[:email]
-    user = User.find_by_email_or_hashed_email enrollment_email
+    user = (User.find_by_email_or_hashed_email enrollment_email) || (User.all.find {|u| u.email_for_enrollments == enrollment_email})
```

Comment:
same concern/question here -- User.all is prohibitively expensive, what breaks if you do not modify this check?

Reply:
Similar to above, I found a fix using the user's application. In this case, `workshop_enrollments_controller.rb` didn't have access to the application so I passed the id from `workshop_enrollment_controller.rb` to `enroll_form.jsx` to this file!